File: 2018/day01.m
fc = load('day01_input.txt'); % frequency changes
n = length(fc);

f = sum(fc); % final frequency
fprintf("Part 1: answer is %d\n", f);

f = 0;                                  % initial frequency
i = 1;                                  % which frequency change to apply
found = false;                          % haven't found a repeat yet
flist = ones(1,1e6)*realmax;            % store all frequencies
ii = 1;                                 % how many frequencies we've stored
fprintf("Part 2: answer is ...");       % it's going to take a while

while (isempty(find(f==flist,1)))       % while we haven't found a repeat,
    flist(ii) = f;                      % store the current frequency
    f = f + fc(i);                      % compute new frequency
    i = mod(i,n) + 1;                   % increment index into fc array
    ii = ii + 1;                        % increment index into flist
end
fprintf(" %d\n", f);

File: 2018/day01.py
# Advent of Code 2018 Day 1

# Part 1
freq = sum([int(line.strip('\n')) for line in open("day01_input.m")])
print('Day 1 Part 1: the frequency is {:d}'.format(freq))

# Part 2
freq = 0
fi = [int(line.strip('\n')) for line in open("day01_input.txt")]
n = len(fi)
i = 0
freqs_found = set()
while ((freq in freqs_found)==False):
	freqs_found.add(freq)
	freq = freq + fi[i]
	i = (i + 1) % n
print('Day 1 Part 2: {:d} is the first freq to repeat'.format(freq))

